ast/src/main/scala/org/opencypher/v9_0/ast/semantics/SemanticExpressionCheck.scala

```
@@ -201,11 +200,18 @@ object SemanticExpressionCheck extends SemanticAnalysisTooling {
       case x:Variable =>
         s => s.ensureVariableDefined(x) match {
           case Right(ss) => SemanticCheckResult.success(ss)
-          case Left(error) => s.declareVariable(x, CTAny.covariant) match {
-            // if the variable is a graph, declaring it will fail
-            case Right(ss) => SemanticCheckResult.error(ss, error)
-            case Left(_error) => SemanticCheckResult.error(s, _error)
-          }
+          case Left(error) =>
+            if (s.declareVariablesToSuppressDuplicateErrors) {
+              // Most of the time we want to suppress if this error occurs again
+              s.declareVariable(x, CTAny.covariant) match {
+                // if the variable is a graph, declaring it will fail
+                case Right(ss) => SemanticCheckResult.error(ss, error)
+                case Left(_error) => SemanticCheckResult.error(s, _error)
+              }
+            } else {
+              // Unless we are ignoring errors anyway in which case this might mess up the scope
```

I guess the `Unless` in this comment refer back to  the earlier comment `Most of the time...`. But if I read only this line I interpreted it as a warning that the else case might mess up the scope. Maybe something like `If we are ignoring errors anyway, suppressing them might mess up the scope` would be clearer?